core/capabilities/gateway_connector/service_wrapper.go

```diff
@@ -0,0 +1,152 @@
+	"slices"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/jonboulle/clockwork"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/services"
+	"github.com/smartcontractkit/chainlink/v2/core/config"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+	gwCommon "github.com/smartcontractkit/chainlink/v2/core/services/gateway/common"
+	"github.com/smartcontractkit/chainlink/v2/core/services/gateway/connector"
+	"github.com/smartcontractkit/chainlink/v2/core/services/gateway/network"
+	"github.com/smartcontractkit/chainlink/v2/core/services/keystore"
+	"github.com/smartcontractkit/chainlink/v2/core/services/keystore/keys/ethkey"
+)
+
+type ServiceWrapper struct {
+	services.StateMachine
+
+	config    config.GatewayConnector
+	keystore  keystore.Eth
+	connector connector.GatewayConnector
+	lggr      logger.Logger
+}
+
+type connectorSigner struct {
+	services.StateMachine
+
+	connector   connector.GatewayConnector
+	signerKey   *ecdsa.PrivateKey
+	nodeAddress string
+	lggr        logger.Logger
+}
+
+var _ connector.Signer = &connectorSigner{}
+
+func NewConnectorSigner(config config.GatewayConnector, signerKey *ecdsa.PrivateKey, lggr logger.Logger) (*connectorSigner, error) {
+	return &connectorSigner{
+		nodeAddress: config.NodeAddress(),
+		signerKey:   signerKey,
+		lggr:        lggr.Named("ConnectorSigner"),
+	}, nil
+}
+
+func (h *connectorSigner) Sign(data ...[]byte) ([]byte, error) {
+	return gwCommon.SignData(h.signerKey, data...)
+}
+
+func (h *connectorSigner) Start(ctx context.Context) error {
+	return h.StartOnce("ConnectorSigner", func() error {
+		return nil
+	})
+}
+func (h *connectorSigner) Close() error {
+	return h.StopOnce("ConnectorSigner", func() (err error) {
+		return nil
+	})
+}
+
+func (h *connectorSigner) SetConnector(connector connector.GatewayConnector) {
+	h.connector = connector
+}
+
+func translateConfigs(f config.GatewayConnector) connector.ConnectorConfig {
+	r := connector.ConnectorConfig{}
+	r.NodeAddress = f.NodeAddress()
+	r.DonId = f.DonID()
+
+	if len(f.Gateways()) != 0 {
+		r.Gateways = make([]connector.ConnectorGatewayConfig, len(f.Gateways()))
+		for index, element := range f.Gateways() {
+			r.Gateways[index] = connector.ConnectorGatewayConfig{Id: element.ID(), URL: element.URL()}
+		}
+	}
+
+	r.WsClientConfig = network.WebSocketClientConfig{HandshakeTimeoutMillis: f.WSHandshakeTimeoutMillis()}
+	r.AuthMinChallengeLen = f.AuthMinChallengeLen()
+	r.AuthTimestampToleranceSec = f.AuthTimestampToleranceSec()
+	return r
+}
+
+// NOTE: this wrapper is needed to make sure that our services are started after Keystore.
+func NewGatewayConnectorServiceWrapper(config config.GatewayConnector, keystore keystore.Eth, lggr logger.Logger) *ServiceWrapper {
+	return &ServiceWrapper{
+		config:   config,
+		keystore: keystore,
+		lggr:     lggr,
+	}
+}
+
+func (e *ServiceWrapper) Start(ctx context.Context) error {
+	return e.StartOnce("GatewayConnectorServiceWrapper", func() error {
+		conf := e.config
```

Comment:
nit: the thing that is being reused is `conf.NodeAddress()`. Can that be set to a variable instead?
Otherwise, should be consistent and use this on line 119